api/v1alpha1/enterprisecontractpolicy_types.go

```diff
@@ -23,46 +23,46 @@ import (
 // Important: Run "make" to regenerate code after modifying this file
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// EnterpriseContractPolicySpec represents the desired state of EnterpriseContractPolicy
+// EnterpriseContractPolicySpec is used to configure the Enterprise Contract Policy
 type EnterpriseContractPolicySpec struct {
-	// Description text describing the the policy or it's intended use
+	// Arbitrary description of the policy or it's intended use
 	// +optional
-	Description *string `json:"description"`
-	// Sources is list of policy sources
+	Description *string `json:"description,omitempty"`
+	// One or more policy sources
 	// +kubebuilder:validation:MinItems:=1
 	Sources []PolicySource `json:"sources"`
-	// Authorization defines a release approval
+	// Authorization defines per component release approvals
 	// +optional
-	Authorization Authorization `json:"authorization"`
+	Authorization *Authorization `json:"authorization,omitempty"`
 	// Exceptions configures exceptions under which the policy is evaluated as successful even if the listed policy checks have reported failure
 	// +optional
 	Exceptions *EnterpriseContractPolicyExceptions `json:"exceptions,omitempty"`
 	// URL of the Rekor instance. Empty string disables Rekor integration
 	// +optional
-	RekorUrl string `json:"rekorUrl"`
+	RekorUrl *string `json:"rekorUrl,omitempty"`
```

Comment:
I prefer having `nil` values for optional fields when they're not present. I find it better to fail with an error or a panic when they're used without testing first. Though I do realize it is a style choice, so I could be persuaded otherwise...

Reply:
I think nil values actually make it easier to use the code because you must perform an additional check to avoid nil pointer de-reference. IMO pointers should only be used if there's a compelling reason to do so.

As a side note, this is a breaking changing which should have, at a bare minimum, its own commit that explains the reason for the change.